src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java

```diff
@@ -0,0 +1,193 @@
+package uk.gov.hmcts.reform.pbis.integration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.microsoft.azure.servicebus.IMessage;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import uk.gov.hmcts.reform.pbis.EmailService;
+import uk.gov.hmcts.reform.pbis.categories.IntegrationTests;
+import uk.gov.hmcts.reform.pbis.model.PrivateBetaRegistration;
+import uk.gov.hmcts.reform.pbis.servicebus.IServiceBusClient;
+import uk.gov.hmcts.reform.pbis.servicebus.IServiceBusClientFactory;
+import uk.gov.hmcts.reform.pbis.servicebus.MessageQueueProcessor;
+import uk.gov.hmcts.reform.pbis.utils.SampleData;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest
+@Category(IntegrationTests.class)
+public class MessageQueueProcessorTest extends AbstractServiceBusTest {
+
+    @Mock
+    private EmailService emailService;
+
+    private MessageQueueProcessor messageQueueProcessor;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        // using the spy here in order to keep track of received messages
+        IServiceBusClientFactory clientFactorySpy = prepareServiceBusClientFactorySpy();
+        messageQueueProcessor = new MessageQueueProcessor(clientFactorySpy, emailService);
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+    }
+
+    @Test
+    public void run_should_read_all_messages_from_subscription() throws Exception {
+        List<PrivateBetaRegistration> registrations = Stream
+            .generate(() -> SampleData.getSampleRegistration(testConfig.getServiceName()))
+            .limit(5)
+            .collect(Collectors.toList());
+
+        for (PrivateBetaRegistration registration : registrations) {
+            serviceBusFeeder.sendMessage(registration);
+        }
+
+        messageQueueProcessor.run();
+
+        // check if there's nothing left to read from the subscription
+        assertThat(receiveMessage()).isNull();
+    }
+
+    @Test
+    public void run_should_call_email_service_for_each_valid_message() throws Exception {
+        int validMesssageCount = 3;
```

Comment:
typo (too many 's')

Reply:
also, perhaps just inline, value can't be changed without breaking the test (the get(0) .. get(2) bit)